File: aliasengine/aliasengine/Data/Ingester/ScriptIngester.swift
//
//  ScriptIngester.swift
//  aliasengine
//
//  Created by Gabriel Coman on 09/06/2022.
//

import Foundation

public class ScriptIngester: FileIngester {
    
    private let alias: String
    private let diskPath: String
    private let fileName: String = ".scripts"
    
    init(withAlias alias: String, andDiskPath path: String) {
        self.alias = alias
        self.diskPath = path
    }
    
    public override func process(fileContents contents: String) -> [IndexItem] {
        return [
            IndexItem(type: .script,
                      name: alias,
                      content: contents,
                      path: fileName,
                      comments: [],
                      pathOnDisk: diskPath)
        ]
    }
}

File: bee/ingester/script_ingester.go
package ingester

import "bee/bbee/models"

// The ScriptIngester just ingests a new full script
type ScriptIngester struct {
	Alias       string
	Path        string
	CurrentTime int64
}

func (s ScriptIngester) Process(content string) []models.IndexItem {
	return []models.IndexItem{
		{
			Name:       s.Alias,
			Content:    content,
			Path:       ".scripts",
			Comments:   []string{},
			PathOnDisk: s.Path,
			Type:       models.ScriptType(models.Script),
			Date:       s.CurrentTime,
		},
	}
}
